platform/view/services/db/driver/sql/init.go

```diff
@@ -0,0 +1,137 @@
+	if err != nil {
+		logger.Error(err)
+		return nil, errors.Wrap(err, "can't open sql database")
+	}
+	table, valid := getTableName(opts.TablePrefix, dataSourceName)
+	if !valid {
+		return nil, errors.Errorf("invalid table name [%s]: only letters and underscores allowed", table)
+	}
+	p := NewPersistence(db, table)
+	if !opts.SchemaExists {
+		if err := p.CreateSchema(); err != nil {
+			return nil, err
+		}
+	}
+	return p, nil
+}
+
+func (d *Driver) NewVersioned(_ view.ServiceProvider, dataSourceName string, config driver.Config) (driver.VersionedPersistence, error) {
+	return d.NewTransactionalVersionedPersistence(nil, dataSourceName, config)
+}
+
+func (d *Driver) New(_ view.ServiceProvider, dataSourceName string, config driver.Config) (driver.Persistence, error) {
+	logger.Infof("opening new unversioned database %s", dataSourceName)
+	opts, err := d.getOps(config)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed getting options for datasource")
+	}
+	db, err := openDB(opts.Driver, opts.DataSource, opts.MaxOpenConns)
+	if err != nil {
+		logger.Error(err)
+		return nil, errors.Wrap(err, "can't open sql database")
+	}
+	table, valid := getTableName(opts.TablePrefix, dataSourceName)
+	if !valid {
+		return nil, errors.Errorf("invalid table name [%s]: only letters and underscores allowed", table)
+	}
+
+	p := NewUnversioned(db, table)
+	if !opts.SchemaExists {
+		if err := p.CreateSchema(); err != nil {
+			return nil, err
+		}
+	}
+	return p, nil
+}
+
+func openDB(driverName, dataSourceName string, maxOpenConns int) (*sql.DB, error) {
+	db, err := sql.Open(driverName, dataSourceName)
+	if err != nil {
+		return nil, err
+	}
+	db.SetMaxOpenConns(maxOpenConns)
+	if err = db.Ping(); err != nil {
+		return nil, err
+	}
+	logger.Infof("connected to [%s,%d]", driverName, maxOpenConns)
+
+	return db, nil
+}
+
+func (d *Driver) getOps(config driver.Config) (Opts, error) {
+	opts := Opts{}
+	if err := config.UnmarshalKey("", &opts); err != nil {
+		return opts, errors.Wrapf(err, "failed getting opts")
+	}
+	if opts.Driver == "" {
+		return opts, errors.New("sql driver not set in core.yaml. See https://github.com/golang/go/wiki/SQLDrivers")
+	}
+	dataSourceOverride := os.Getenv(EnvVarKey)
+	if dataSourceOverride != "" {
+		logger.Infof("overriding datasource with from env var [%s] ([%d] characters)", len(dataSourceOverride), EnvVarKey)
+		opts.DataSource = dataSourceOverride
+	}
+	if opts.DataSource == "" {
+		return opts, errors.Errorf("either the dataSource in core.yaml or %s environment variable must be set to a dataSource that can be used with the %s golang driver", EnvVarKey, opts.Driver)
+	}
+	if opts.TablePrefix == "" {
+		opts.TablePrefix = "fsc"
+	}
+	return opts, nil
+}
+
+func getTableName(prefix, name string) (table string, valid bool) {
+	table = fmt.Sprintf("%s_%s", prefix, name)
+	r := regexp.MustCompile("^[a-zA-Z_]+$")
+	return table, r.MatchString(name)
+}
+
+func init() {
+	db.Register("sql", &Driver{})
```

Comment:
I am wondering if this should happen as part of the `sdk.go` as it is responsible to orchestrate all the drivers needed. Having the `init` function without the use of build flags comes with the risk that we lose track on where we load certain code.

Reply:
This is how it's done everywhere in FSC and token sdk... @adecaro might have an opinion?

Reply:
I think we will discuss this in the near future given that we are discussing the introduction of dependency injection frameworks. 
Right now, we use the same approach that is used by the `sql` package in the Go sdk.